### PR TITLE
Prepare release notes and files for v0.6.0

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Prepare release notes and files for v0.6.0

#### Which issue(s) this PR fixes:

Part of #1729

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-0.6.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+## v0.5.3
```
Comment (tenzen-y) [inline CHANGELOG/CHANGELOG-0.6.md:1]:
```suggestion
## v0.6.0
```

**Review by tenzen-y — COMMENTED** [review]
Just a nit.

File: CHANGELOG/CHANGELOG-0.6.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  The status of the internal cache and queues is also logged on demand when a SIGUSR2 is sent to kueue, regardless of the log level. (#1528, @alculquicondor)
+- Changing tolerations in an inadmissible job triggers an admission retry with the updated tolerations. (#1304, @stuton)
+- Increase the default number of reconcilers for Pod and Workload objects to 5, each. (#1589, @alculquicondor)
+- Jobs preserve their position in the queue if the number of pods change before being admitted (#1223, @yaroslava-serdiuk)
+- Make the image build setting CGO_ENABLED configurable (#1391, @anishasthana)
+- MultiKueue: Add live status updates for multikueue JobSets (#1668, @trasc)
+- MultiKueue: Support for JobSets. (#1606, @trasc)
+- Support RayCluster as a queue-able workload in Kueue (#1520, @vicentefb)
+- Support for retry of provisioning request.
+
```
Comment (tenzen-y) [inline CHANGELOG/CHANGELOG-0.6.md:43]:
```suggestion
```

ditto.

File: CHANGELOG/CHANGELOG-0.6.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+- Avoid recreating a Workload for a finished Job and finalize a job when the workload is declared finished. (#1383, @achernevskii)
+- Do not (re)create ProvReq if the state of admission check is Ready (#1617, @mimowo)
+- Fix Kueue crashing at the log level 6 when re-admitting workloads (#1644, @mimowo)
+- Fix a bug in the pod integration that unexpected errors will occur when the pod isn't found (#1512, @achernevskii)
+- Fix a bug that plain pods managed by kueue will remain in a terminating state, due to a finalizer (#1342, @tenzen-y)
+- Fix client-go libraries bug that can not operate clusterScoped resources like ClusterQueue and ResourceFlavor. (#1294, @tenzen-y)
+- Fix fungibility policy `Preempt` where it was not able to utilize the next flavor if preemption was not possible. (#1366, @alculquicondor)
+- Fix handling of preemption within a cohort when there is no borrowingLimit. In that case,
+  during preemption, the permitted resources to borrow were calculated as if borrowingLimit=0, instead of unlimited.
+
```
Comment (tenzen-y) [inline CHANGELOG/CHANGELOG-0.6.md:66]: ditto.

File: CHANGELOG/CHANGELOG-0.6.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+- Fix client-go libraries bug that can not operate clusterScoped resources like ClusterQueue and ResourceFlavor. (#1294, @tenzen-y)
+- Fix fungibility policy `Preempt` where it was not able to utilize the next flavor if preemption was not possible. (#1366, @alculquicondor)
+- Fix handling of preemption within a cohort when there is no borrowingLimit. In that case,
+  during preemption, the permitted resources to borrow were calculated as if borrowingLimit=0, instead of unlimited.
+  
+  As a consequence, when using `reclaimWithinCohort`, it was possible that a workload, scheduled to ClusterQueue with no borrowingLimit, would preempt more workloads than needed, even though it could fit by borrowing. (#1561, @mimowo)
+- Fix the synchronization of the admission check state based on recreated ProvisioningRequest (#1585, @mimowo)
+- Fixed fungibility policy `whenCanPreempt: Preempt`. The admission should happen in the flavor for which preemptions were issued. (#1332, @alculquicondor)
+- Kueue replicas are advertised as Ready only once the webhooks are functional.
+
```
Comment (tenzen-y) [inline CHANGELOG/CHANGELOG-0.6.md:71]: ditto.

File: CHANGELOG/CHANGELOG-0.6.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+### Feature
+
+- Add Prebuilt Workload support for JobSets. (#1575, @trasc)
+- Add events for transitions of the provisioning AdmissionCheck (#1271, @stuton)
+- Add prebuilt workload support for batch/job. (#1358, @trasc)
+- Add support for groups of plain Pods. (#1319, @achernevskii)
+- Allow configuring featureGates on helm charts. (#1314, @B1F030)
+- At log level 6, the usage of ClusterQueues and cohorts is included in logs.
+
```
Comment (tenzen-y) [inline CHANGELOG/CHANGELOG-0.6.md:33]: @alculquicondor Can we remove this empty line?

Comment (alculquicondor) [reply CHANGELOG/CHANGELOG-0.6.md:33]: It's intentional. This is how it renders:

image

**Pull request merged**